fireapp/lib/domain/models/unavailability/unavailability_time.dart

```diff
@@ -7,7 +7,7 @@ part 'unavailability_time.g.dart';
 @freezed
 class UnavailabilityTime with _$UnavailabilityTime{
   const factory UnavailabilityTime({
-    required int eventId,
+    required int? eventId,
```

Comment:
I would add a comment here explaining that eventId is only ever null when creating a new instance (will tell future devs that it is safe to force not-null when getting a server response)